jdaviz/configs/mosviz/plugins/tools.py

```diff
@@ -18,12 +19,43 @@ def _is_matched_viewer(self, viewer):
         return isinstance(viewer, (MosvizProfile2DView, MosvizProfileView))
 
     def _map_limits(self, from_viewer, to_viewer, limits={}):
+        components = self.viewer.state.data_collection[0]._components
+        # Determine cid for spectral axis
+        cid = None
+        for key in components.keys():
+            if 'Wavelength' in str(key):
+                cid = str(key)
+                break
+            elif 'Wave' in str(key):
+                cid = str(key)
+                break
+
+        if cid is not None:
+            native_unit = u.Unit(self.viewer.state.data_collection[0].get_component(cid).units)
+        else:
+            native_unit = ''
```

Comment:
We could condense this a bit (but existing is fine too if you'd rather keep it)

```suggestion
        for key in components.keys():
            strkey = str(key)
            if 'Wavelength' in strkey or 'Wave' in strkey:
                native_unit = u.Unit(self.viewer.state.data_collection[0].get_component(strkey).units)
                break
        else:
            # no matches found
            native_unit = ''
```

Reply:
I was thinking too that casting each check was overkill, will be included in the next commit!